### PR TITLE
Parse and set bullet solver iterations

### PR DESCRIPTION
**Pull request by iche033**
# 🎉 New feature

## Summary

Added support for users to configure the bullet solver iterations. 

It parses the `//world/physics/bullet/solver/iters` SDF param and sets that value to the physics engine using the new Solver feature, `SetSolversIterations`, introduced in https://github.com/gazebosim/gz-physics/pull/609.  

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/LevelManager.cc [hunk 1]
```diff
@@ -119,26 +119,39 @@ void LevelManager::ReadLevelPerformerInfo()
...
-      auto solver =
-          dartElem->GetElement("solver")->Get<std::string>("solver_type");
-
-      this->runner->entityCompMgr.CreateComponent(worldEntity,
-          components::PhysicsSolver(solver));
+      if (auto solverElem = bulletElem->FindElement("solver"))
+      {
+        if (solverElem->HasElement("iters"))
+        {
+          uint32_t solverIterations = solverElem->Get<uint32_t>("iters");
```
Comment (ahcorde) [inline src/LevelManager.cc:150]: include `<cstdint>`

File: src/SdfEntityCreator.cc [hunk 2]
```diff
@@ -305,26 +305,39 @@ Entity SdfEntityCreator::CreateEntities(const sdf::World *_world)
...
-      auto solver =
-          dartElem->GetElement("solver")->Get<std::string>("solver_type");
-
-      this->dataPtr->ecm->CreateComponent(worldEntity,
-          components::PhysicsSolver(solver));
+      if (auto solverElem = bulletElem->FindElement("solver"))
+      {
+        if (solverElem->HasElement("iters"))
+        {
+          uint32_t solverIterations = solverElem->Get<uint32_t>("iters");
```
Comment (ahcorde) [inline src/SdfEntityCreator.cc:336]: include `<cstdint>`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/SdfEntityCreator.cc (hunk 2 above)
Comment (iche033) [reply src/SdfEntityCreator.cc:338]: added. [35bce70](https://github.com/gazebosim/gz-sim/pull/2351/commits/35bce70ef3bf4df77f2b3e790f2c22e7ee43006c)

File: src/LevelManager.cc (hunk 1 above)
Comment (iche033) [reply src/LevelManager.cc:151]: added. [35bce70](https://github.com/gazebosim/gz-sim/pull/2351/commits/35bce70ef3bf4df77f2b3e790f2c22e7ee43006c)

File: include/gz/sim/components/Physics.hh [hunk 3]
```diff
@@ -65,6 +65,12 @@ namespace components
       class PhysicsSolverTag, serializers::StringSerializer>;
   GZ_SIM_REGISTER_COMPONENT("gz_sim_components.PhysicsSolver",
        PhysicsSolver)
+
+  /// \brief The number of solver iterations for each step.
+  using PhysicsSolverIterations = Component<uint32_t,
```
[the remark this answers is not in the document]
Comment (iche033) [reply include/gz/sim/components/Physics.hh:71]: added. [35bce70](https://github.com/gazebosim/gz-sim/pull/2351/commits/35bce70ef3bf4df77f2b3e790f2c22e7ee43006c)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**